cmd/api/src/api/auth.go

```diff
@@ -337,55 +339,92 @@ func SetSecureBrowserCookie(request *http.Request, response http.ResponseWriter,
 	var (
-		hostURL    = *ctx.FromRequest(request).Host
-		requestCtx = request.Context()
+		hostURL      = *ctx.FromRequest(request).Host
+		requestCtx   = request.Context()
+		commitID     uuid.UUID
+		user         model.User
+		err          error
+		loginRequest = LoginRequest{
+			Username: principalNameOrEmail,
+		}
+
+		loginOutcome = model.AuditLogStatusFailure
+		loginError   error
 	)
 
-	if user, err := s.db.LookupUser(requestCtx, principalNameOrEmail); err != nil {
+	// Set LoginAttempt method for audit logs
+	switch authProvider.(type) {
+	case model.SAMLProvider:
+		loginRequest.LoginMethod = auth.ProviderTypeSAML
+	case model.OIDCProvider:
+		loginRequest.LoginMethod = auth.ProviderTypeOIDC
+	}
+
+	// Generate commit ID for audit logging
+	if commitID, err = uuid.NewV4(); err != nil {
+		log.Errorf("Error generating commit ID for login: %s", err)
+		WriteErrorResponse(requestCtx, BuildErrorResponse(http.StatusInternalServerError, "audit log creation failure", request), response)
+		return
+	}
+
+	// Log the intent to authenticate
+	s.auditLogin(requestCtx, commitID, user, loginRequest, model.AuditLogStatusIntent, nil)
+
+	// Log authentication success or failure
+	defer func() {
+		s.auditLogin(requestCtx, commitID, user, loginRequest, loginOutcome, loginError)
+	}()
+
+	if user, err = s.db.LookupUser(requestCtx, principalNameOrEmail); err != nil {
 		if !errors.Is(err, database.ErrNotFound) {
 			HandleDatabaseError(request, response, err)
 		} else {
-			WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
+			WriteErrorResponse(requestCtx, BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
 		}
-	} else {
-		switch typedAuthProvider := authProvider.(type) {
-		case model.SAMLProvider:
-			if !user.SAMLProviderID.Valid || typedAuthProvider.ID != user.SAMLProvider.ID {
-				WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
-				return
-			}
-		case model.OIDCProvider:
-			//todo connect to db provider table
-
-			// Delete pre-auth cookies regardless
-			DeleteBrowserCookie(request, response, AuthPKCECookieName)
-			DeleteBrowserCookie(request, response, AuthStateCookieName)
-			break
-		case model.AuthSecret:
-			WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusBadRequest, "invalid auth provider", request), response)
-			return
-		default:
-			WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusBadRequest, "invalid auth provider", request), response)
+		loginError = err
+		return
+	}
+
+	switch typedAuthProvider := authProvider.(type) {
+	case model.SAMLProvider:
+		if !user.SAMLProviderID.Valid || typedAuthProvider.ID != user.SAMLProvider.ID {
+			loginError = ErrorUserNotAuthorizedForProvider
+			WriteErrorResponse(requestCtx, BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
```

Comment:
It may be too large of a refactor to consider "in-scope", however, you may want to consider hoisting this check out of the switch-case and assert the `SSOProvider`'s ID matches the user's.

``` go
func (s authenticator) CreateSSOSession(request *http.Request, response http.ResponseWriter, principalNameOrEmail string, ssoProvider SSOProvider) {
    // ...
    if ssoProvider.ID != user.SSOProviderID {
        loginError = ErrorUserNotAuthorizedForProvider
        WriteErrorResponse(requestCtx, BuildErrorResponse(http.StatusForbidden, "user is not allowed", request), response)
    }
    // ...
}

Reply:
This isn't possible until after BED-4945 is completed as that id column is not maintained outside of the original backfill migration.